### PR TITLE
emacs: Add option to install version 28.2

### PR DESCRIPTION
**Pull request by pranith**

### CODE REVIEW

**Comment by tldahlgren** [comment]
Please resolve the conflict.  If there are any single quotes for strings after that, they'll need to be fixed too since the style/`black` checks will complain.

File: var/spack/repos/builtin/packages/emacs/package.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ class Emacs(AutotoolsPackage, GNUMirrorPackage):
     gnu_mirror_path = "emacs/emacs-24.5.tar.gz"
 
     version('master', branch='master')
+    version('28.2', sha256='a6912b14ef4abb1edab7f88191bfd61c3edd7085e084de960a4f86485cb7cad8')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/emacs/package.py]: Confirmed the sha256.

**Review by tldahlgren — COMMENTED** [review]
Confirmed the sha256.  

PR needs conflicts resolved and remaining strings with single quotes fixed.

**Comment by pranith** [comment]
Updated as required. Please squash all the changes before merging. Thanks!

**Review by bernhardkaindl — APPROVED** [review]
emacs-28.2 builds

**Pull request merged**